src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,100 @@
+package duke.storage;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.DukeIoException;
+import duke.tasks.TaskList;
+import duke.tasks.TaskType;
+import duke.ui.Ui;
+
+/**
+ * The storage is used to save and load tasks from the local disk.
+ */
+public class Storage {
+
+    /** The path to the save file. */
+    private final String path;
+
+    /**
+     * Creates a new Storage object.
+     *
+     * @param path The path to the save file.
+     * @throws DukeException If there is an error creating the save file.
+     */
+    public Storage(String path) throws DukeException {
+        this.path = path;
+        File file = new File(path);
```

Comment:
Based on your code in Duke.java, I thought you were planning on using the java.nio package(which from what I read, is newer and better than java.io which is what this line is using. 

I actually tried using Paths entirely initially but I couldn't find many tutorials online for nio classes so I stuck to using only java.io.File classes. 

Since you already incorporated some nio classes like Path and Files into your code maybe you can consider using those mainly instead?

You can try taking a look at [this](https://www.baeldung.com/java-how-to-create-a-file)